pkg/processor/procbuiltin/func_wrapper.go

```diff
@@ -38,23 +38,25 @@ func NewFuncWrapper(f func(context.Context, record.Record) (record.Record, error
 
 	return FuncWrapper{
 		f:       f,
-		inInsp:  inspector.New(log.New(zl), 1000),
-		outInsp: inspector.New(log.New(zl), 1000),
+		inInsp:  inspector.New(log.New(zl), inspector.DefaultBufferSize),
+		outInsp: inspector.New(log.New(zl), inspector.DefaultBufferSize),
 	}
 }
 
 func (f FuncWrapper) Process(ctx context.Context, inRec record.Record) (record.Record, error) {
 	// todo same behavior as in procjs, probably can be enforced
 	f.inInsp.Send(ctx, inRec)
 	outRec, err := f.f(ctx, inRec)
-	f.outInsp.Send(ctx, outRec)
+	if err == nil {
+		f.outInsp.Send(ctx, outRec)
+	}
```

Comment:
Nitpick: I'd prefer keeping the happy flow to the left, do you mind turning the condition around? 🙏

Reply:
Good one!